geps/gep-91/index.md

```diff
@@ -1,22 +1,92 @@
-* Status: Provisional
+* Status: Implementable 
 
 (See definitions in [GEP Status][/contributing/gep#status].)
 
 ## TLDR
 
 This GEP proposes a way to validate the TLS certificate presented by the downstream client to the server
-(Gateway Listener in this case) during a [TLS Handshake Protocol][], also commonly referred to as mutual TLS (mTLS).
+(Gateway Listener in this case) during a [TLS Handshake Protocol][].
 
 ## Goals
-- Define an API field to specify the CA Certificate within the Gateway Listener configuration that can be used as a trusted anchor to validate the certificates presented by the client.
+- Define an API field to specify the CA Certificate within the Gateway Listener configuration that can be used as a trust anchor to validate the certificates presented by the client. This use case has been been highlighted in the [Gateway API TLS Use Cases][] document under point 7.
 
 ## Non-Goals
 - Define other fields that can be used to verify the client certificate such as the Certificate Hash or Subject Alt Name. 
 
+### API
+
+* Introduce a `clientValidation` field of type `ClientValidationContext` within [GatewayTLSConfig][] that can be used to validate the client initiating the TLS connection
+to the Gateway.
+* Introduce a `caCertificateRefs` field within `ClientValidationContext` that can be used to specify a list of CA Certificates that
+can be used as a trust anchor to validate the certificates presented by the client.
+
+#### GO
+
+```go
+// ClientValidationContext holds configuration that can be used to validate the client intiating the TLS connection
+// to the Gateway.
+// By default, no client specific configuration is validated.
+type ClientValidationContext struct {
+    // CACertificateRefs contains one or more references to
+    // Kubernetes objects that contain TLS certificates of
+    // the Certificate Authorities that can be used to
+    // as a trust anchor to validate the certificates presented by the client.
+    //
+    // A single CACertRef to a Kubernetes ConfigMap kind has "Core" support.
+    // Implementations MAY choose to support attaching multiple CA certificates to
+    // a Listener, but this behavior is implementation-specific.
+    //
+    // Support: Core - An optional single reference to a Kubernetes ConfigMap,
+    // with the CA certificate in a key named `ca.crt`.
+    //
+    // Support: Implementation-specific (More than one reference, or other kinds
+    // of resources).
+    //
+    // References to a resource in different namespace are invalid UNLESS there
+    // is a ReferenceGrant in the target namespace that allows the certificate
+    // to be attached. If a ReferenceGrant does not allow this reference, the
+    // "ResolvedRefs" condition MUST be set to False for this listener with the
+    // "RefNotPermitted" reason.
+    //
+    // +kubebuilder:validation:MaxItems=8
+    // +optional
+    CACertificateRefs []corev1.ObjectReference `json:”caCertificateRefs,omitempty”`
```

Comment:
Do we really want to use `corev1.ObjectReference` here? Seems that it should be either `SecretObjectReference` (same as GatewayTLSConfig.CertificateRefs) or `ConfigMapObjectReference` (same as gep-1897 BackendTLSPolicyConfig).

Reply:
pros for using `corev1.ObjectReference`
* this reference can be a `ConfigMap` with a `ca.crt` key or a `Secret` with a `ca.crt` (which can be the same secret that also has `tls.crt` and `tls.key` used in `CertificateRefs` )
* reference can live in a different namespace and `ResourceGrant` can be used to authorize the reference (similar to `CertificateRefs` )